q2_diversity_lib/tests/test_alpha.py

```diff
@@ -39,77 +39,47 @@ def test_non_phylogenetic_passed_empty_table(self):
-            BIOMV210Format(empty_table_fp, mode='r')
-        input_table_fp = self.get_data_path('faith_test_table.biom')
-        self.input_table_as_BIOMV210Format = \
-            BIOMV210Format(input_table_fp, mode='r')
-        rf_table_fp = self.get_data_path('faith_test_table_rf.biom')
-        self.rf_table_as_BIOMV210Format = BIOMV210Format(rf_table_fp, mode='r')
-        pa_table_fp = self.get_data_path('faith_test_table_pa.biom')
-        self.pa_table_as_BIOMV210Format = BIOMV210Format(pa_table_fp, mode='r')
-
-        empty_tree_fp = self.get_data_path('empty.tree')
-        self.empty_tree_as_NewickFormat = NewickFormat(empty_tree_fp, mode='r')
-        input_tree_fp = self.get_data_path('faith_test.tree')
-        self.input_tree_as_NewickFormat = NewickFormat(input_tree_fp, mode='r')
-        root_only_tree_fp = self.get_data_path('root_only.tree')
-        self.root_only_tree_as_NewickFormat = \
-            NewickFormat(root_only_tree_fp, mode='r')
-        missing_tip_tree_fp = self.get_data_path('missing_tip.tree')
-        self.missing_tip_tree_as_NewickFormat = \
-            NewickFormat(missing_tip_tree_fp, mode='r')
-
+        self.fn = self.plugin.actions['faith_pd']
+        self.tbl = self.artifact('FeatureTable[Frequency]',
+                                 'faith_test_table.biom')
+        self.tre = self.artifact('Phylogeny[Rooted]', 'faith_test.tree')
         self.expected = pd.Series({'S1': 0.5, 'S2': 0.7, 'S3': 1.0,
                                    'S4': 100.5, 'S5': 101},
                                   name='faith_pd')
 
     def test_receives_empty_table(self):
-        # empty table generated from self.empty_table with biom v2.1.7
-        empty_table = self.empty_table_as_BIOMV210Format
+        table = self.artifact('FeatureTable[Frequency]', 'empty_table.biom')
         with self.assertRaisesRegex(ValueError, 'empty'):
-            faith_pd(table=empty_table,
-                     phylogeny=self.input_tree_as_NewickFormat)
+            self.fn(table=table, phylogeny=self.tre)
 
     def test_method(self):
-        actual = faith_pd(table=self.input_table_as_BIOMV210Format,
-                          phylogeny=self.input_tree_as_NewickFormat)
-        pdt.assert_series_equal(actual, self.expected)
+        actual_art, = self.fn(table=self.tbl, phylogeny=self.tre)
+        self.assert_vec_equal(actual_art, self.expected)
 
     def test_accepted_types_have_consistent_behavior(self):
-        freq_table = self.input_table_as_BIOMV210Format
-        rel_freq_table = self.rf_table_as_BIOMV210Format
-        p_a_table = self.pa_table_as_BIOMV210Format
-        accepted_tables = [freq_table, rel_freq_table, p_a_table]
-        for table in accepted_tables:
-            actual = faith_pd(table=table,
-                              phylogeny=self.input_tree_as_NewickFormat)
-            pdt.assert_series_equal(actual, self.expected)
-
-    def test_passed_emptytree(self):
```

Comment:
What happens in this case now? Or is that a test you're still writing?

Reply:
Good question - this case was never actually possible, because the format prevents it:

```python
In [1]: from qiime2 import Artifact

In [2]: Artifact.import_data('Phylogeny[Rooted]', 'q2_diversity_lib/tests/data/empty.tree')
<frozen importlib._bootstrap>:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 216 from C header, got 232 from PyObject
<frozen importlib._bootstrap>:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 216 from C header, got 232 from PyObject
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
Input In [2], in <cell line: 1>()
----> 1 Artifact.import_data('Phylogeny[Rooted]', 'q2_diversity_lib/tests/data/empty.tree')

File ~/.conda/envs/q2dev/lib/python3.8/site-packages/qiime2/sdk/result.py:278, in Artifact.import_data(cls, type, view, view_type)
    275     format_ = view_type
    277 provenance_capture = archive.ImportProvenanceCapture(format_, md5sums)
--> 278 return cls._from_view(type_, view, view_type, provenance_capture,
    279                       validate_level='max')

File ~/.conda/envs/q2dev/lib/python3.8/site-packages/qiime2/sdk/result.py:306, in Artifact._from_view(cls, type, view, view_type, provenance_capture, validate_level)
    303 recorder = provenance_capture.transformation_recorder('return')
    304 transformation = from_type.make_transformation(to_type,
    305                                                recorder=recorder)
--> 306 result = transformation(view, validate_level)
    308 if type_raw in pm.validators:
    309     validation_object = pm.validators[type]

File ~/.conda/envs/q2dev/lib/python3.8/site-packages/qiime2/core/transform.py:68, in ModelType.make_transformation.<locals>.transformation(view, validate_level)
     66 def transformation(view, validate_level='min'):
     67     view = self.coerce_view(view)
---> 68     self.validate(view, level=validate_level)
     70     new_view = transformer(view)
     72     new_view = other.coerce_view(new_view)

File ~/.conda/envs/q2dev/lib/python3.8/site-packages/qiime2/core/transform.py:143, in FormatType.validate(self, view, level)
    140     raise TypeError("%r is not an instance of %r."
    141                     % (view, self._view_type))
    142 # Formats have a validate method, so defer to it
--> 143 view.validate(level)

File ~/.conda/envs/q2dev/lib/python3.8/site-packages/qiime2/plugin/model/file_format.py:35, in _FileFormat.validate(self, level)
     33 elif hasattr(self, 'sniff'):
     34     if not self.sniff():
---> 35         raise ValidationError("%s is not a(n) %s file"
     36                               % (self.path, self.__class__.__name__))
     38 # TODO: define an abc.abstractmethod for `validate` when sniff is
     39 # removed instead of this
     40 else:
     41     raise NotImplementedError("%r does not implement validate."
     42                               % type(self))

ValidationError: q2_diversity_lib/tests/data/empty.tree is not a(n) NewickFormat file
```